util/chplenv/chpl_gpu.py

```diff
@@ -114,5 +127,24 @@ def get_runtime():
             return chpl_gpu_runtime
     return get()
 
+def validateLlvmBuiltForRocm():
+    exists, returncode, my_stdout, my_stderr = utils.try_run_command(
+        [chpl_llvm.get_llvm_config(), "--targets-built"])
+
+    if not exists or returncode != 0:
+        return False
+
+    targets = my_stdout.strip().split(" ")
+    return "AMDGPU" in targets
+
+
+@memoize
 def validate(chplLocaleModel, chplComm):
-    pass
+    if chplLocaleModel != "gpu":
+        return True
+
+    if get() == 'rocm':
+        if not validateLlvmBuiltForRocm():
+            error("LLVM not built for AMDGPU, consider setting CHPL_LLVM to 'bundled'")
```

Comment:
Don't we also use LLVM's cuda target? Perhaps we should also validate that the LLVM is built for cuda when that's in use? Seems to be easy enough to generalize `validateLlvmBuiltForRocm`.

Reply:
Good point, I'll add it a check for that

Reply:
Updated in 0493a4bdef99245b760c4d2929a553ef2eeebeb1